src/NServiceBus.Transport.SQS/InputQueuePump.cs

```diff
@@ -405,12 +456,14 @@ async Task DeleteMessage(Message message, string s3BodyKey, CancellationToken me
             try
             {
                 // should not be canceled
-                await sqsClient.DeleteMessageAsync(inputQueueUrl, message.ReceiptHandle, messageProcessingCancellationToken).ConfigureAwait(false);
+                await sqsClient
+                    .DeleteMessageAsync(inputQueueUrl, message.ReceiptHandle, messageProcessingCancellationToken)
+                    .ConfigureAwait(false);
             }
-            catch (ReceiptHandleIsInvalidException ex)
+            catch (Exception ex) when (!ex.IsCausedBy(messageProcessingCancellationToken))
```

Comment:
I think we want to keep this as a `ReceiptHandleIsInvalidException` `catch`, but tweak the wording and the `messagesToBeDeleted` stuff.

Looks like we should also move the `s3BodyKey` `if` block to be inside the inside the `try`.